synapse/tests/test_lib_grammar.py

```diff
@@ -685,6 +685,25 @@
     'reverse(inet:ipv4)',
     'reverse(inet:ipv4=1.2.3.4)',
     'reverse(*$form=$valu)',
+    'test:str=foobar -> inet:dns*',
+    'test:str=foobar -> inet:dns:*',
+    'test:str=foobar -> (edge:has, inet:dns:a)',
+    'test:str=foobar -> (edge:has, inet:dns*)',
+    'test:str=foobar -> $foo',
+    'test:str=foobar -+> inet:dns*',
+    'test:str=foobar -+> inet:dns:*',
+    'test:str=foobar -+> (edge:has, inet:dns:a)',
+    'test:str=foobar -+> (edge:has, inet:dns*)',
+    'test:str=foobar -+> $foo',
+    'test:str=foobar -(refs)> inet:dns:*',
+    'inet:fqdn=foo.com :zone -> (edge:has, inet:dns:a)',
+    'inet:fqdn=foo.com :zone -> $foo',
+    'inet:fqdn=foo.com :zone -+> (edge:has, inet:dns:a)',
+    'inet:fqdn=foo.com :zone -+> $foo',
+    'test:*#foo',
+    'test:*#foo@=2016',
+    'test:*#foo:lol',
+    'test:*#foo:lol=20',
```

Comment:
do we need grammar tests for wild-card form filters in here? `-inet:*` etc

Reply:
Yeah i'll add some more grammar tests here